### PR TITLE
3768 - Fix error saving field-based page

### PR DESCRIPTION
**Pull request by WillNigel23**

### CODE REVIEW

File: app/models/transcription_field.rb [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
   has_many :table_cells
-  has_many :spreadsheet_columns, -> { order 'position' }, :dependent => :destroy
-  validates :options, presence: true, if: Proc.new {|field| field.input_type == 'select'}, on: [:create, :update]
+  has_many :spreadsheet_columns, -> { order 'position' }, dependent: :destroy
 
+  validates :options, presence: true, if: Proc.new { |field| field.input_type == 'select' }, on: [:create, :update]
   validates :percentage, numericality: { allow_nil: true, greater_than: 0, less_than_or_equal_to: 100 }
   validates :page_number, numericality: { allow_nil: true, greater_than: 0, less_than_or_equal_to: 1000 }
-
+  validates :label, format: { without: /[\[\]\{\}]/, message: "cannot contain '{', '}', '[' or ']'" }
```
Comment (WillNigel23) [inline app/models/transcription_field.rb:14]: Validation for checking characters that might be interpreted as hash like '{}' or '[]'

File: app/views/transcription_field/_field_layout.html.slim [hunk 2]
```diff
@@ -14,32 +14,17 @@
...
-                -options = field.options.split(";") unless field.options.nil?
-                =select_tag("fields[#{field.id}][#{field.label}]", options_for_select(options, content), class: 'field-input')
-              -elsif field.input_type == "description"
-                =hidden_field_tag "fields[#{field.id}][#{field.label}]", content, class: 'field-input'
-              -elsif field.input_type == "alt text"
-                =text_area_tag "fields[#{field.id}][#{field.label}]", content, class: 'field-input alt-text', maxlength: '255'
-                div.character-count
-              -else
-                =text_area_tag "fields[#{field.id}][#{field.label}]", content, class: 'field-input'
+              =generate_field_input(field, cell)
```
Comment (WillNigel23) [inline app/views/transcription_field/_field_layout.html.slim:21]: I moved some of the logic here to the helper function.

File: app/helpers/transcription_field_helper.rb [hunk 3]
```diff
@@ -22,4 +24,35 @@ def field_layout(array)
     @field_array = array.zip(@values)
   end
 
+  def generate_field_input(field, cell)
+    input_name = formatted_field_name(field)
+
+    label = label_tag(field.label.parameterize, field.label.titleize)
```
Comment (WillNigel23) [inline app/helpers/transcription_field_helper.rb:30]: I titleize the label for the field inputs. Let me know if it should not be like this after all

**Comment by WillNigel23** [comment]
Moving back to in progress. seems there are a lot of specs failing due to changed name.

File: app/models/transcription_field.rb (hunk 1 above)
Comment (benwbrum) [reply app/models/transcription_field.rb:14]: Let's remove the `{}` validation -- only the hash element accessor `[]` is used by Rails to create structured parameters, `{}` isn't a problem.  (I just tested that on `development` and `{}` works.

File: app/views/transcription_field/_field_layout.html.slim (hunk 2 above)
Comment (benwbrum) [reply app/views/transcription_field/_field_layout.html.slim:21]: Great!

File: app/helpers/transcription_field_helper.rb (hunk 3 above)
Comment (benwbrum) [reply app/helpers/transcription_field_helper.rb:30]: Let's not titleize the label -- many of our customers have strong opinions about this.

**Review by benwbrum — CHANGES_REQUESTED** [review]
Remove titleize and permit curly-braces.

**Pull request merged**